lib/msf/core/post/file.rb

```diff
@@ -501,19 +501,22 @@ def rm_rf(*remote_dirs)
   alias :dir_rm :rm_rf
 
   #
-  # Rename a remote file.
+  # Renames a remote file and returns true on success and false
+  # on failure
   #
   # @param old_file [String] Remote file name to move
   # @param new_file [String] The new name for the remote file
   def rename_file(old_file, new_file)
+    verification_token = Rex::Text.rand_text_alphanumeric(8)
     if session.type == "meterpreter"
       return (session.fs.file.mv(old_file, new_file).result == 0)
```

Comment:
This is raising a `Rex::Post::Meterpreter::RequestError` when the `old_file` doesn't exist. This error should be caught and cause `false` to be returned. This is probably better than checking `exists?` in this case but the exception will also be raised if there's a permissions issue with the old or new files.

Reply:
This will also fail if `new_file` is a directory. The shell versions however will just move the file into the directory. That's at least the case for Windows's `move /y ...` and Posix's `mv -f ...` commands.

For consistency, this should check if `new_file` is a directory and if so, concatenate the filename of `old_file` to the `new_file` directory.

Reply:
should we rename the method to `move_file`?

Reply:
Done with these changes :+1: , sorry for the delay!